registry/storage/driver/gcs/gcs.go

```diff
@@ -852,21 +810,22 @@ func (d *driver) Walk(ctx context.Context, path string, f storagedriver.WalkFn,
 	return storagedriver.WalkFallback(ctx, d, path, f, options...)
 }
 
-func startSession(client *http.Client, bucket string, name string) (uri string, err error) {
+func (w *writer) newSession() (uri string, err error) {
 	u := &url.URL{
 		Scheme:   "https",
 		Host:     "www.googleapis.com",
-		Path:     fmt.Sprintf("/upload/storage/v1/b/%v/o", bucket),
-		RawQuery: fmt.Sprintf("uploadType=resumable&name=%v", name),
+		Path:     fmt.Sprintf("/upload/storage/v1/b/%v/o", w.driver.bucket),
+		RawQuery: fmt.Sprintf("uploadType=resumable&name=%v", w.key),
+	}
+	req, err := http.NewRequest(http.MethodPost, u.String(), nil)
```

Comment:
Needs more ~cowbell~ context. You know the drill.
```suggestion
	req, err := http.NewRequestWithContext(ctx, http.MethodPost, u.String(), nil)
```